Modules/Messages/test/Messages.Domain.Tests/Messages/DecomposeForTests.cs

```diff
@@ -0,0 +1,180 @@
+        if (output.R2IsAnonymized)
+            recipient2.Address.Should().Be(ANONYMIZED_ADDRESS);
+        else
+            recipient2.Address.Should().NotBe(ANONYMIZED_ADDRESS);
+    }
+}
+
+public enum Participant
+{
+    Sender,
+    Recipient1,
+    Recipient2
+}
+
+public record TestInput
+{
+    public TestInput(int id,
+        Participant decomposer,
+        Participant relationshipTo,
+        bool senderHasAlreadyDecomposedRelationshipWithRecipient1,
+        bool recipient1HasAlreadyDecomposedRelationshipWithSender,
+        bool senderHasAlreadyDecomposedRelationshipWithRecipient2,
+        bool recipient2HasAlreadyDecomposedRelationshipWithSender)
+    {
+        Id = id;
+
+        Message = TestData.CreateMessageWithTwoRecipients();
+
+        Decomposer = decomposer switch
+        {
+            Participant.Sender => Message.CreatedBy,
+            Participant.Recipient1 => Message.Recipients.First().Address,
+            Participant.Recipient2 => Message.Recipients.Second().Address,
+            _ => throw new Exception()
+        };
+
+        if (Decomposer == Message.CreatedBy)
+        {
+            Peer = relationshipTo switch
+            {
+                Participant.Recipient1 => Message.Recipients.First().Address,
+                Participant.Recipient2 => Message.Recipients.Second().Address,
+                _ => throw new Exception()
+            };
+        }
+        else
+            Peer = Message.CreatedBy;
+
+        if (recipient1HasAlreadyDecomposedRelationshipWithSender)
+            Message.DecomposeFor(Message.Recipients.First().Address, Message.CreatedBy, DecomposeForTests.ANONYMIZED_ADDRESS);
+        if (recipient2HasAlreadyDecomposedRelationshipWithSender)
+            Message.DecomposeFor(Message.Recipients.Second().Address, Message.CreatedBy, DecomposeForTests.ANONYMIZED_ADDRESS);
+        if (senderHasAlreadyDecomposedRelationshipWithRecipient1)
+            Message.DecomposeFor(Message.CreatedBy, Message.Recipients.First().Address, DecomposeForTests.ANONYMIZED_ADDRESS);
+        if (senderHasAlreadyDecomposedRelationshipWithRecipient2)
+            Message.DecomposeFor(Message.CreatedBy, Message.Recipients.Second().Address, DecomposeForTests.ANONYMIZED_ADDRESS);
+    }
+
+
+    public int Id { get; init; }
+    public Message Message { get; }
+    public IdentityAddress Decomposer { get; }
+    public IdentityAddress Peer { get; }
+}
+
+// ReSharper disable InconsistentNaming
+public record TestOutput(bool R1_HiddenForSender, bool R1_HiddenForRecipient, bool R2_HiddenForSender, bool R2_HiddenForRecipient, bool SenderIsAnonymized, bool R1IsAnonymized, bool R2IsAnonymized);
+// ReSharper restore InconsistentNaming
+
+public class TestDataWithAllCases : TheoryData<TestInput, TestOutput>
+{
+    public TestDataWithAllCases()
+    {
+        Add(new TestInput(0, Participant.Sender, Participant.Recipient1, false, false, false, false), new TestOutput(true, false, false, false, false, false, false));
```

Comment:
Yes we do. I didn't do it here because the lines are already quite lengthy, and adding the parameter names makes them even longer. Do you think it would improve readability? If so, I will add the names.

Reply:
It would definitely improve readability.
You might get away with adding it to a single line if preferred, since they are all quite similar.

Reply:
The following picture shows what it looks like if I would add the parameters for all lines:

![image](https://github.com/user-attachments/assets/39da165c-1594-4a79-8381-6c007bb45e0a)

If we only had it for the first, it would look as follows:
![image](https://github.com/user-attachments/assets/759d148b-1745-4e6b-afe5-e8fe2201843e)

I would probably rely on the inline hints displayed by the IDE in this case, which you can show if necessary:
![image](https://github.com/user-attachments/assets/8eb0a2d1-8cce-45af-8238-5316b71b968a)


But if you still think it makes sense to use named parameters, I'm totally okay with adding them :)

Reply:
You're right, becomes too massive.
Don't add them, it's fine.